### PR TITLE
fix: change design of mention list item

### LINKED ISSUE #12705: Tag Table in discussion get wrong if multiple table have the same name
CC @chirag-madlani 

**Is your feature request related to a problem? Please describe.**
Related to the discussion feature. How about when tagging a specific table with '#' but may table has the same name? how could we pick the right table on the right schema we want?
image

**Describe the solution you'd like**
Change the way to show the list of tables when tagged in the discussion feature.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by karanh37**
### Describe your changes:

Fixes #12705 

I worked on ... because ...

image

Screenshot 2023-08-14 at 5 13 37 PM

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Comment by harshach** [comment]
@karanh37 @Sachin-chaurasiya @chirag-madlani I've asked many times to display the user profile picture while doing @ . We have closed those tickets but its never works. Can we take a look why its not working or we didn't ship it

**Review by harshach — DISMISSED** [review]

**Review by Sachin-chaurasiya — DISMISSED** [review]

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/cypress/common/common.js [hunk 1]
```diff
@@ -500,35 +495,37 @@ export const visitEntityDetailsPage = (
...
+      ) {
+        // if term is available in search suggestion, redirecting to entity details page
+        cy.get(`[data-testid="${id}"] [data-testid="data-name"]`)
+          .should('be.visible')
+          .first()
+          .click();
+      } else {
+        // if term is not available in search suggestion,
+        // hitting enter to search box so it will redirect to explore page
+        cy.get('body').click(1, 1);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/common/common.js:513]:
```suggestion
// to close popup
  cy.clickOutside();
```

File: openmetadata-ui/src/main/resources/ui/cypress/common/common.js [hunk 2]
```diff
@@ -500,35 +495,37 @@ export const visitEntityDetailsPage = (
...
+  cy.wait('@searchQuery').then(() => {
+    cy.wait(500);
+    cy.get('body').then(($body) => {
+      // checking if requested term is available in search suggestion
+      if (
+        $body.find(`[data-testid="${id}"] [data-testid="data-name"]`).length
+      ) {
+        // if term is available in search suggestion, redirecting to entity details page
+        cy.get(`[data-testid="${id}"] [data-testid="data-name"]`)
+          .should('be.visible')
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/common/common.js:507]: avoid click after should be visible and use it only for assertion

File: openmetadata-ui/src/main/resources/ui/cypress/common/common.js [hunk 3]
```diff
@@ -500,35 +495,37 @@ export const visitEntityDetailsPage = (
...
+          .click();
+      } else {
+        // if term is not available in search suggestion,
+        // hitting enter to search box so it will redirect to explore page
+        cy.get('body').click(1, 1);
+        cy.get('[data-testid="searchBox"]').type('{enter}');
+        verifyResponseStatusCode('@explorePageSearch', 200);
+
+        cy.get(`[data-testid="${entity}-tab"]`).should('be.visible').click();
+        cy.get(`[data-testid="${entity}-tab"]`).should('be.visible');
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/common/common.js:518]: same avoid click on should be visible

File: openmetadata-ui/src/main/resources/ui/cypress/common/common.js [hunk 4]
```diff
@@ -500,35 +495,37 @@ export const visitEntityDetailsPage = (
...
 
-  verifyResponseStatusCode('@getEntityDetails', 200);
-  cy.get('body').click(1, 1);
-  cy.get('[data-testid="searchBox"]').clear();
+        cy.get(`[data-testid="${id}"]`).scrollIntoView().click();
+      }
+    });
+
+    verifyResponseStatusCode('@getEntityDetails', 200);
+    cy.get('body').click(1, 1);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/common/common.js:526]:
```suggestion
    // to close popup
  cy.clickOutside();
```

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**